pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller_test.go

```diff
@@ -0,0 +1,185 @@
+package clustermemberremovalcontroller
+
+import (
+	"context"
+	"testing"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+
+	configv1 "github.com/openshift/api/config/v1"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	machinelistersv1beta1 "github.com/openshift/client-go/machine/listers/machine/v1beta1"
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfakeclient "k8s.io/client-go/kubernetes/fake"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestClusterMemberRemovalController(t *testing.T) {
+	alwaysTrueIsFunctionalMachineAPIFn := func() (bool, error) { return true, nil }
+
+	scenarios := []struct {
+		name                             string
+		isFunctionalMachineAPIFn         func() (bool, error)
+		initialObjectsForConfigMapLister []runtime.Object
+		initialObjectsForNodeLister      []runtime.Object
+		initialObjectsForMachineLister   []runtime.Object
+		initialEtcdMemberList            []*etcdserverpb.Member
+		validateFn                       func(t *testing.T, fakeEtcdClient etcdcli.EtcdClient)
+	}{
+		// scenario 1
+		{
+			name:                             "happy path: an etcd member has a corresponding machine and node resources",
+			isFunctionalMachineAPIFn:         alwaysTrueIsFunctionalMachineAPIFn,
+			initialObjectsForConfigMapLister: []runtime.Object{wellKnownEtcdEndpointsConfigMap()},
+			initialObjectsForNodeLister:      []runtime.Object{wellKnownMasterNode()},
+			initialObjectsForMachineLister:   []runtime.Object{wellKnownMasterMachine()},
```

Comment:
Do these tests make sense if there's no validate function? You're running a test against something and not checking what action was taken after `.sync` was called it seems? Or have I missed something

Reply:
fair point, the test should check if the member wasn't removed.